core/src/main/java/com/scalar/db/api/LikeExpression.java

```diff
@@ -0,0 +1,157 @@
+package com.scalar.db.api;
+
+import com.google.common.base.MoreObjects;
+import com.scalar.db.io.TextColumn;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class LikeExpression extends ConditionalExpression {
+  private static final String DEFAULT_ESCAPE_CHAR = "\\";
+  private final String escape;
+  private final String regexPattern;
+
+  /**
+   * Constructs a {@code LikeExpression} with the specified column and operator. For the escape
+   * character, the default one ("\", i.e., backslash) is used.
+   *
+   * @param column a target column used to compare
+   * @param operator an operator used to compare the target column
+   */
+  LikeExpression(TextColumn column, Operator operator) {
+    this(column, operator, DEFAULT_ESCAPE_CHAR);
+  }
+
+  /**
+   * Constructs a {@code LikeExpression} with the specified column, operator and escape character.
+   * The escape character must be a string of a single character or an empty string. If an empty
+   * string is specified, the escape character is disabled.
+   *
+   * @param column a target column used to compare
+   * @param operator an operator used to compare the target column
+   * @param escape an escape character for the like operator
+   */
+  LikeExpression(TextColumn column, Operator operator, String escape) {
+    super(column, operator);
+    if (operator != Operator.LIKE && operator != Operator.NOT_LIKE) {
+      throw new IllegalArgumentException(
+          "Operator must be like or not-like. Operator: " + operator);
+    }
+    if (escape == null || escape.length() > 1) {
+      throw new IllegalArgumentException(
+          "Escape character must be a string of a single character or an empty string");
+    }
+    this.escape = escape;
+    this.regexPattern =
+        convertRegexPatternFrom(
+            column.getTextValue(), escape.length() == 0 ? null : escape.charAt(0));
+  }
+
+  /**
+   * Validate and convert SQL 'like' pattern to a Java regular expression. Underscores (_) are
+   * converted to '.' and percent signs (%) are converted to '.*', other characters are quoted
+   * literally. If an escape character specified, escaping is done for '_', '%', and the escape
+   * character itself. An invalid pattern will throw {@code IllegalArgumentException}. This method
+   * is implemented referencing the following Spark SQL implementation.
+   * https://github.com/apache/spark/blob/a8eadebd686caa110c4077f4199d11e797146dc5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala
+   *
+   * @param likePattern a SQL LIKE pattern to convert
+   * @param escape an escape character.
+   * @return the equivalent Java regular expression of the given pattern
+   */
+  private String convertRegexPatternFrom(String likePattern, @Nullable Character escape) {
```

Comment:
Can we move this logic into `Snapshot` since it's only used there?

Reply:
The regular expression is only used in `Snapshot`, so technically, yes. But I think we should check if a specified pattern is an expected one because throwing an exception with "pattern must not ..." from `Snapshot` is weird. So, the choices are 1) keeping this as is and 2) only adding the check here. IMHO, 1) is not so bad since the check and conversion are tightly coupled with. What do you think?
p.s. `isMatchedWith` can be moved to `Snapshot` in any case.

Reply:
I think we should choose the `2)` option. We should separate the regular expression thing from `LikeExpression` because the responsibility of the regular expression thing is not for `LikeExpression` but Consensus Commit (e.g., `JdbcTransaction` doesn't use the regular expression). What do you think?

Reply:
> because the responsibility of the regular expression thing is not for LikeExpression but Consensus Commit (e.g., JdbcTransaction doesn't use the regular expression).

Right. Totally agreed. Thanks for pointing it out!